product_configurator/wizard/product_configurator.py

```diff
@@ -278,16 +308,23 @@ def apply_onchange_values(self, values, field_name, field_onchange):
         if not cfg_vals:
             cfg_vals = self.value_ids
 
-        field_type = type(field_name)
-
+        field_type = type(field_names)
         field_prefix = self._prefixes.get("field_prefix")
         custom_field_prefix = self._prefixes.get("custom_field_prefix")
-        if field_type == list or (
-            not field_name.startswith(field_prefix)
-            and not field_name.startswith(custom_field_prefix)
+        domain_field_prefix = self._prefixes.get("domain_field_prefix")
+        local_field_name = field_names and field_names[0].startswith(field_prefix)
+        local_custom_field = field_names and field_names[0].startswith(
+            custom_field_prefix
+        )
+        local_domain_prefix = field_names and field_names[0].startswith(
+            domain_field_prefix
+        )
+        if field_type == list and (
```

Comment:
The previous lines of code kind of assume the `field_names`  is a `list`
(with things like `field_names and field_names[0].startswith`)

So, can it be something different than list or not?
Either this condition is redundant, or the previous code might fail

Looking at the core methods, it seems it's safe to assume it's always going to be a list, and so this condition can be removed

For simplicity and readability, I'd do something like this:

```py
regular_field_names = (
    fname for fname in field_names
    if not any(fname.startswith(prefix) for prefix in self._prefixes.values())
)
if regular_field_names:
```

Reply:
In odoo v17.0, it will return the list of fields changed from the web client. In the case of form view, even if it is always one field, it is still a list of a field.

Reply:
Yes, that's my point.
Checking `if field_type == list` is redundant then :)

Reply:
Solved

Reply:
Thanks!